blockdag/dag.go

```diff
@@ -1799,6 +1813,79 @@ func (dag *BlockDAG) SubnetworkID() *subnetworkid.SubnetworkID {
 	return dag.subnetworkID
 }
 
+// AddDelayedBlock adds a block to the delayed blocks list.
+func (dag *BlockDAG) AddDelayedBlock(block *util.Block, delay time.Duration) {
+	dag.delayedBlocksLock.Lock()
+	defer dag.delayedBlocksLock.Unlock()
+
+	processTime := dag.timeSource.AdjustedTime().Add(delay)
+	log.Debugf("Adding block to delayed blocks queue (block hash: %s, process time: %s)", block.Hash().String(), processTime)
+	db := &delayedBlock{
+		block:       block,
+		processTime: processTime,
+	}
+
+	dag.delayedBlocks[*block.Hash()] = db
+	dag.delayedBlocksQueue.Push(db)
+}
+
+// ProcessDelayedBlocks loops over all delayed blocks and processes blocks which are due.
+// This method is invoked after processing a block (ProcessBlock method).
+func (dag *BlockDAG) ProcessDelayedBlocks() error {
+	dag.delayedBlocksLock.Lock()
+	defer dag.delayedBlocksLock.Unlock()
+	// Check if the delayed block with the earliest process time should be processed
+	for dag.delayedBlocksQueue.Len() > 0 {
+		earliestDelayedBlockProcessTime := dag.delayedBlocksQueue.peek().processTime
+		if dag.timeSource.AdjustedTime().After(earliestDelayedBlockProcessTime) {
+			delayedBlock := dag.popDelayedBlock()
+			dag.dagLock.Unlock()
+			_, _, err := dag.ProcessBlock(delayedBlock.block, BFAfterDelay)
+			dag.dagLock.Lock()
+			if err != nil {
+				log.Errorf("Error while processing delayed block (block %s)", delayedBlock.block.Hash().String())
+				return err
+			}
+			log.Debugf("Processed delayed block (block %s)", delayedBlock.block.Hash().String())
+		} else {
```

Comment:
Please use the opposite condition to break, and then you can drop the `else` condition